tests/48admin.pl

```diff
@@ -1,5 +1,22 @@
 use Future::Utils qw( repeat );
 
+
+# poll the status endpoint until it completes. Returns the final status.
+sub await_purge_complete {
+   my ( $admin_user, $purge_id ) = @_;
+
+   return repeat( sub {
+      do_request_json_for( $admin_user,
+         method  => "GET",
+         uri     => "/r0/admin/purge_history_status/$purge_id",
+      )->then( sub {
+         my ($body) = @_;
+         assert_json_keys( $body, "status" );
+         Future->done( $body->{status} );
+      })
+   }, while => sub { $_[0]->get eq 'active' });
```

Comment:
I'd use either `repeat_until_true` or `retry_until_success` helper functions, as they have a small backoff

Reply:
well, I'm deliberately doing my own repeat so that I can distinguish between failure (request fails) and not ready yet (status == 'active').

I've added a backoff.

Reply:
Ah ok, I just assumed one of those could be used. Ah well